### PR TITLE
Issue #SB-2157 feat: editors and unit testing

### PR DESCRIPTION
**Pull request by revathitarento**
Issue #SB-2157 feat: editors and unit testing

### CODE REVIEW

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 1]
```diff
@@ -81,13 +98,37 @@ export class UserService {
       }
     );
   }
+/**
+    * method to fetch appId and Ekstep_env from server.
+    */
+    public getAppidEnv(): void {
+      const url = `get/envData`;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: should be kept in config

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 2]
```diff
@@ -43,12 +44,28 @@ export class UserService {
...
+ */
+  public _appId: string;
+  /**
+   * Reference of channel
+   */
+  public channel: string;
+  /**
+   * Reference of dims
+   */
+  public dims: Array<string> = [];
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: dims not assigned value in user service but referred in many components

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 3]
```diff
@@ -43,12 +44,28 @@ export class UserService {
    * reference of lerner service.
    */
   public learner: LearnerService;
+  /**
+ * Reference of appId
+ */
+  public _appId: string;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: this should be private, get method is created to access this

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 4]
```diff
@@ -43,12 +44,28 @@ export class UserService {
...
+  /**
+   * Reference of channel
+   */
+  public channel: string;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: this should be private, get method is created to access this

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 5]
```diff
@@ -43,12 +44,28 @@ export class UserService {
...
+  /**
+   * Reference of dims
+   */
+  public dims: Array<string> = [];
+  /**
+   * Reference of Ekstep_env
+   */
+  public _env: string;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: this should be private, get method is created to access this

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 6]
```diff
@@ -81,13 +98,37 @@ export class UserService {
       }
     );
   }
+/**
+    * method to fetch appId and Ekstep_env from server.
+    */
+    public getAppidEnv(): void {
+      const url = `get/envData`;
+      this.http.get(url).subscribe(res => {
+        this._appId = res['appId'];
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: this._appId = res.appId; use this, keep consistance

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts [hunk 7]
```diff
@@ -81,13 +98,37 @@ export class UserService {
...
+        this._env = res['ekstep_env'];
+      });
+    }
+
+    /**
+     * get method to fetch appId.
+     */
+    get appId(): any {
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: return type should be string

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 8]
```diff
@@ -93,11 +98,13 @@ export class CreateCollectionComponent implements OnInit {
          this.userProfile = user.userProfile;
         }
       });
-    this.showLoader = false;
     this.userForm = this.formBuilder.group({
       name: '',
       description: ''
     });
+    if (document.getElementById('collectionEditor')) {
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: why we need to remove 'collectionEditor', which is present in a different route. Which should be destroyed when route changes

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 9]
```diff
@@ -125,18 +132,23 @@ generateData() {
...
+      'loaderMessage': this.resourceService.messages.stmsg.m0016,
+  };
     this.editorService.create(requestData).subscribe(res => {
       this.showLoader = false;
-      const type = 'CollectionEditor';
-      const state = '';
+      const type = 'Collection';
+      const state = 'state';
       const framework = 'framework';
+      modalCollection.approve();
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: why we need to close model, angular will destroy this model when we navigate to different url

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 10]
```diff
@@ -125,18 +132,23 @@ generateData() {
...
+    this.loaderMessage = {
+      'loaderMessage': this.resourceService.messages.stmsg.m0016,
+  };
     this.editorService.create(requestData).subscribe(res => {
       this.showLoader = false;
-      const type = 'CollectionEditor';
-      const state = '';
+      const type = 'Collection';
+      const state = 'state';
       const framework = 'framework';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: framework api should be called to fetch the value, hardcoding string is not accepted.

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-study-material/create-study-material.component.ts [hunk 11]
```diff
@@ -1,13 +1,183 @@
...
+  *  Create colletion creates the content Id
+  */
+  createContent() {
+    const state = 'state';
+    const requestData = {
+      content: this.generateData()
+    };
+    this.editorService.create(requestData).subscribe(res => {
+      this.showLoader = true;
+      if (res && res.responseCode === 'OK') {
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-study-material/create-study-material.component.ts]: no need to check this

File: src/app/migration/src/src/app/modules/workspace/components/create-content/create-content.component.ts [hunk 12]
```diff
@@ -27,5 +28,9 @@ export class CreateContentComponent {
   constructor(resourceService: ResourceService) {
     this.resourceService = resourceService;
   }
+  ngOnInit() {
+   this.showLoader = false;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/create-content/create-content.component.ts]: why show loader is required

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.spec.ts [hunk 13]
```diff
@@ -1,25 +1,101 @@
...
+    spyOn(editorService, 'getById').and.returnValue(Observable.of(mockRes.successResult));
+    component.openCollectionEditor();
+    const rspData = mockRes.successResult.result.content;
+    component.validateRequest(rspData, mockRes.validateModal);
+
+    component.updateModeAndStatus('draft');
+    component.updateModeAndStatus('live');
+    component.updateModeAndStatus('review');
+    component.updateModeAndStatus('flagged');
+
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.spec.ts]: there no assertion here

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.spec.ts [hunk 14]
```diff
@@ -1,25 +1,101 @@
...
+    fixture.detectChanges();
+
+
+    spyOn(editorService, 'getById').and.returnValue(Observable.of(mockRes.successResult));
+    component.openCollectionEditor();
+    const rspData = mockRes.successResult.result.content;
+    component.validateRequest(rspData, mockRes.validateModal);
+
+    component.updateModeAndStatus('draft');
+    component.updateModeAndStatus('live');
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.spec.ts]: what are we doing by calling the same method with different param

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts [hunk 15]
```diff
@@ -1,15 +1,632 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, AfterViewInit, NgZone } from '@angular/core';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts]: why do we need NgZone

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts [hunk 16]
```diff
@@ -1,15 +1,632 @@
...
+import { Injectable } from '@angular/core';
+import * as _ from 'lodash';
+import * as $ from 'jquery';
+import * as  iziModal from 'izimodal/js/iziModal';
+import { ResourceService, ConfigService, ToasterService, ServerResponse, IUserData, IUserProfile } from '@sunbird/shared';
+import { UserService } from '@sunbird/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { CustomWindow } from './../../../interfaces/custom.window';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts]: import { CustomWindow } from './../../../interfaces'

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts (hunk 2 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: done

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts (hunk 5 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: When copying files from other branch to new branch i missed to update appid and channelid, now i added

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts (hunk 4 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 9 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: Modal remains opened even after we navigate to other router. so we have to close modal, also i am displaying the loader message.

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts (hunk 15 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts]: We are using onClosing function to close the modal, the angular function written for this jquery plugin event is not executed. so using ngZone we can execute our angular code inside the jquery event.

File: src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts (hunk 16 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/create-editors/collection-editor/collection-editor.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/create-content/create-content.component.ts (hunk 12 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/create-content/create-content.component.ts]: not required, so removed.

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 10 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: As per rajath, it is required, he will be taking the data from the api and update it, which he had done in Angular 1.

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 17]
```diff
@@ -125,18 +132,23 @@ generateData() {
...
     };
+    this.loaderMessage = {
+      'loaderMessage': this.resourceService.messages.stmsg.m0016,
+  };
     this.editorService.create(requestData).subscribe(res => {
       this.showLoader = false;
-      const type = 'CollectionEditor';
-      const state = '';
+      const type = 'Collection';
+      const state = 'state';
```
[the remark this answers is not in the document]
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: Yes, the default routing state is "Draft" after collection is created. Removed.

File: src/app/migration/src/src/app/modules/core/services/user/user.service.ts (hunk 6 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/core/services/user/user.service.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-study-material/create-study-material.component.ts (hunk 11 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-study-material/create-study-material.component.ts]: removed

**Pull request merged**